api/core/v1alpha1/common_helpers.go

```diff
@@ -228,3 +233,35 @@ func setDefaultServiceArgs(object interface{}) {
 		return
 	}
 }
+
+func ClaimedBucket(c client.Client, ls *LogSet) (*BucketClaim, error) {
+	uniqLabel := UniqueBucketLabel(ls)
+	bcList := &BucketClaimList{}
+	if err := c.List(context.TODO(), bcList, client.InNamespace(ls.Namespace), client.MatchingLabels{BucketUniqLabel: uniqLabel}); err != nil {
+		return nil, err
+	}
+
+	switch len(bcList.Items) {
+	case 0:
+		return nil, nil
+	case 1:
+		return &bcList.Items[0], nil
+	default:
+		return nil, fmt.Errorf("list more than one buckets for logset: %s/%s", ls.Namespace, ls.Name)
+	}
+}
+
+func UniqueBucketLabel(ls *LogSet) string {
+	s3Provider := ls.Spec.SharedStorage.S3
```

Comment:
This may be nil.

Reply:
The caller must ensure S3 is not nil. The first thing when writing some bucket logic is to check whether S3 is nil, if S3 is nil, We should do nothing.